### ISSUE TITLE
'str_node' object has no attribute 'get'

### ISSUE THREAD

**Issue by kgeisink** [awslabs/cfn-python-lint#209]
*cfn-lint version: (`cfn-lint --version`)*
0.3.5

*Description of issue.*
Note: The original problem is just me being stupid writing my template, but the result of the linter was an unexpected error.

I found this issue when using the [CodePipeline Stages Actions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codepipeline-pipeline-stages-actions.html) resource, specifically the [OutputArtifacts](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codepipeline-pipeline-stages-actions-outputartifacts.html) property. 

The correct way to specify this property is 
```
OutputArtifacts:
  - Name: SourceCodeAsZip
```

but I accidentally wrote

```
OutputArtifacts:
  - SourceCodeAsZip
```

Which resulted in the following error:
`2018-07-13 12:15:39,408 - cfnlint - ERROR - Tried to process rules on file cloudformation/template.yaml but got an error: 'str_node' object has no attribute 'get'`

It is basically a missing property kind of thing, but I assume that since it is part of a list it messes up the parsing somehow? I haven't gone deep enough into the workings of the linter to know for sure.

I compared it to a resource property that is similar but without the list ([CodeBuild Source](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codebuild-project-source.html)) (if property Type has value CODEPIPELINE it is the only required property for Source) and in this case the linter provides the following error:

`E3002 Expecting an object at Resources/CodeBuildProject/Properties/Source
cloudformation/template.yaml:123:7`

Maybe something similar can be used?

**Comment by cmmeyer** [comment]
Thanks for the detailed report. The CodePipeline spec isn't actually part of the CloudFormation resource specification, so this was hand coded. I like your proposed solution -- I think we need some better try logic in the rule.  I'll flag this as a bug.

**Comment by fatbasstard** [comment]
Checking this now. It's apparantly not a bug in the CodePipeline rule, but in the actual parsing itself:

```
  File "/Users/frankvanboven/Documents/dev/cfn-python-lint/src/cfnlint/__init__.py", line 196, in resource_property
    value, resource_type, resource_spec_property.get('ItemType')
  File "/Users/frankvanboven/Documents/dev/cfn-python-lint/src/cfnlint/__init__.py", line 196, in resource_property
    value, resource_type, resource_spec_property.get('ItemType')
  File "/Users/frankvanboven/Documents/dev/cfn-python-lint/src/cfnlint/__init__.py", line 184, in resource_property
    for resource_property, resource_property_value in properties.items():
AttributeError: 'str_node' object has no attribute 'items'
```

When I specify 
```
              OutputArtifacts:
                - MyApp
```

To be continued...

**Comment by kddejong** [comment]
@kgeisink can you check v0.4.1 and see if this is resolved?

**Comment by kgeisink** [comment]
@kddejong It works like a charm! 🎉 

```
E3002 Expecting an object at Resources/Pipeline/Properties/Stages/0/Actions/0/OutputArtifacts/0
cloudformation/pipeline.yaml:219:15
```
